Upgrade Guide (#851)
spatialos/UnrealGDK: SpatialGDK/Documentation/content/upgrading.md

<%(TOC)%>
# Keep your GDK up to date

> To use the SpatialOS GDK for Unreal, you need two git repositories:<br>

Reviewer: ```suggestion
To use the SpatialOS GDK for Unreal, you need software from two git repositories:<br>
```

> * [The SpatialOS Unreal Engine fork.](https://github.com/improbableio/UnrealEngine)

Reviewer: ```suggestion
* [The SpatialOS Unreal Engine fork](https://github.com/improbableio/UnrealEngine)
```

* [The GDK](https://github.com/spatialos/UnrealGDK)<br>

> 

Reviewer: ```suggestion
You download bith od these as part of the [Get started]({{urlRoot}}/content/get-started/introduction) steps. <br/>
```

> To benefit from the latest version of GDK, always develop your game on the latest version. Whenever you update your GDK, you **must** also update your SpatialOS Unreal Engine fork. If you don't, errors can be caused by the two repositories being out of sync.

Reviewer: ```suggestion
It is good practice to always develop your game on the latest version of the software by updating the software. This ensures you benefit from the most up-to-date functionality. Whenever you update your GDK software, you **must** also update your SpatialOS Unreal Engine fork software. If you don't, you may get errors from them being out of synch.
```

> We recommend that you update your version of the GDK and SpatialOS Unreal Engine fork every week.

Reviewer: ```suggestion
We recommend that you update your version of the GDK and SpatialOS Unreal Engine fork every week.  To do this, follow the steps below.
```

> It is good practice to always develop your game on the latest version of the software by regularly updating it. This ensures you benefit from the most up-to-date functionality. Whenever you update your GDK software, you **must** also update your SpatialOS Unreal Engine fork software. If you don't, you may get errors from them being out of synch.

Reviewer: ```suggestion
To ensure you benefit from the most up-to-date functionality, always develop your game on the latest version of the software by regularly updating it. Whenever you update your GDK software, you **must** also update your SpatialOS Unreal Engine fork software. If you don't, you might get errors from them being out of synch.
```

## Step 1: Ensure you're on the release branches.

> If you followed our [get started]({{urlRoot}}/content/get-started/introduction) guide, you have these repositories cloned on your computer.<br>

Reviewer: ```suggestion
If you followed our [Get started]({{urlRoot}}/content/get-started/introduction) guide, you have these repositories cloned on your computer.<br>
```

> * Your `UnrealGDK` repository should have the `release` branch checked out.<br>

Reviewer: ```suggestion
* Your `UnrealEngine` repository should have the branch ending with `-SpatialOSUnrealGDK-release`checked out.<br>
```

> * Your `UnrealEngine` repository should have the branch ending with `-SpatialOSUnrealGDK-release`checked out.<br>

Reviewer: ```suggestion
* Your `UnrealGDK` repository should have the `release` branch checked out.<br>
```

> You can determine which branch is checked out by following the instructions below:<br>

Reviewer: ```suggestion
You can find out which branch you have checked out by following the instructions below:<br>
```
